packages/framework/aqueduct/src/data-objects/treeDataObject.ts

```diff
@@ -0,0 +1,131 @@
+	 * it's also impossible for this single view schema to handle multiple different stored schema.
+	 * Therefore, with this current API, two different applications (or different versions of the same application)
+	 * with differing stored schema requirements (as implied by their view schema) can not collaborate on the same tree.
+	 * The only schema evolution that's currently possible is upgrading the schema to one that supports a superset of what the old schema allowed,
+	 * and collaborating between clients which have view schema that exactly correspond to that stored schema.
+	 * Future work on tree as well as these utilities should address this limitation.
+	 */
+	public abstract readonly config: TreeViewConfiguration<TSchema>;
+
+	/**
+	 * The underlying tree.
+	 */
+	#tree: ITree | undefined;
+
+	/**
+	 * View of the underlying tree.
+	 */
+	#treeView: TreeView<TSchema> | undefined;
+
+	/**
+	 * Gets the underlying {@link @fluidframework/tree#ITree}.
+	 * @throws If the tree has not yet been initialized, this will throw an error.
+	 */
+	private get initializedTree(): ITree {
+		if (!this.#tree) {
+			// Note: Can't use `UsageError` because adding dependency on `telemetry-utils` would create a cycle.
+			// TODO: would probably be useful to move our shared error types in a more accessible location.
+			throw new Error(this.getUninitializedErrorString("root"));
+		}
+
+		return this.#tree;
+	}
+
+	/**
+	 * Gets the view of the underlying {@link @fluidframework/tree#ITree}.
+	 * @throws If the tree has not yet been initialized, this will throw an error.
+	 */
+	public get tree(): TreeView<TSchema> {
+		if (this.#treeView === undefined) {
+			throw new Error(this.getUninitializedErrorString("tree"));
+		}
+		return this.#treeView;
+	}
+
+	public override async initializeInternal(existing: boolean): Promise<void> {
+		if (existing) {
+			// data store has a root tree so we just need to set it before calling initializingFromExisting
+			const channel = await this.runtime.getChannel(treeChannelId);
+			if (SharedTree.is(channel)) {
+				this.#tree = channel;
+			} else {
+				throw new Error(
+					`Content with id ${channel.id} is not a SharedTree and cannot be loaded with treeDataObject.`,
+				);
+			}
+		} else {
+			this.#tree = SharedTree.create(this.runtime, treeChannelId);
+			(this.#tree as unknown as ISharedObject).bindToContext();
+		}
+
+		await super.initializeInternal(existing);
+	}
+
+	protected override async initializingFirstTime(): Promise<void> {
+		this.#treeView = this.initializedTree.viewWith(this.config);
+
+		// Initialize the tree content and schema.
+		this.#treeView.initialize(this.createInitialTree());
+	}
+
+	protected override async initializingFromExisting(): Promise<void> {
+		this.#treeView = this.initializedTree.viewWith(this.config);
+	}
+
+	protected override async hasInitialized(): Promise<void> {
+		if (this.#treeView === undefined) {
+			throw new Error(this.getUninitializedErrorString("tree"));
+		}
```

Comment:
these methods are really for users, we should do all our initialization within initialize internal. i know users can call super, and maybe they should, but the existing dataobject doesn't require it, and we should go for parity, even though we don't necessarily like pure data object

Reply:
I'll play around with this some more.